svgbasic.go

```diff
@@ -200,36 +201,113 @@ type SVGBasicType struct {
 	Segments [][]SVGBasicSegmentType
 }
 
+// This function parse float and its unit. The result is converted in pt
+// wich is the default document value. The function return the scale
+// to apply to each position or distance value.
+func parseFloatWithUnit(val string) (float64, float64, error) {
+	var adjustToPt float64
+	var removeUnitChar int
+	var floatValue float64
+	var err error
+
+	if strings.HasSuffix(val, "pt") {
+		removeUnitChar = 2
+		adjustToPt = 1.0
+	} else if strings.HasSuffix(val, "in") {
+		removeUnitChar = 2
+		adjustToPt = 72.0
+	} else if strings.HasSuffix(val, "mm") {
+		removeUnitChar = 2
+		adjustToPt = 72.0 / 25.4
+	} else if strings.HasSuffix(val, "cm") {
+		removeUnitChar = 2
+		adjustToPt = 72.0 / 2.54
+	} else if strings.HasSuffix(val, "pc") {
+		removeUnitChar = 2
+		adjustToPt = 12.0
+	} else { // default is pixel
+		removeUnitChar = 0
+		adjustToPt = 1.0 / 96.0
+	}
```

Comment:
Go "good practices" recommend to use a `switch`.
```suggestion
	switch {
	case strings.HasSuffix(val, "pt"):
		removeUnitChar = 2
		adjustToPt = 1.0
	case strings.HasSuffix(val, "in"):
		removeUnitChar = 2
		adjustToPt = 72.0
	case strings.HasSuffix(val, "mm"):
		removeUnitChar = 2
		adjustToPt = 72.0 / 25.4
	case strings.HasSuffix(val, "cm"):
		removeUnitChar = 2
		adjustToPt = 72.0 / 2.54
	case strings.HasSuffix(val, "pc"):
		removeUnitChar = 2
		adjustToPt = 12.0
	default: // default is pixel
		removeUnitChar = 0
		adjustToPt = 1.0 / 96.0
	}
```

perhaps also add a case for `"px"` pixel units ?